packages/scene-composer/src/components/panels/AddComponentMenu.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+import { defineMessages, MessageDescriptor, useIntl } from 'react-intl';
+import * as THREE from 'three';
+import { IconProps } from '@awsui/components-react';
+
+import { ToolbarItemOptionRaw, ToolbarItemOptions } from '../toolbars/common/types';
+import { sceneComposerIdContext } from '../../common/sceneComposerIdContext';
+import { IDataOverlayComponentInternal, useStore } from '../../store';
+import { findComponentByType } from '../../utils/nodeUtils';
+import { COMPOSER_FEATURES, KnownComponentType } from '../../interfaces';
+import { ToolbarItem } from '../toolbars/common/ToolbarItem';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { Component } from '../../models/SceneModels';
+import { IDataBindingComponentInternal } from '../../store/internalInterfaces';
+
+interface AddComponentMenuProps {
+  onSelect?: (selectedObject: ObjectTypes) => void;
+}
+
+// Note: ObjectType String is used to record metric. DO NOT change existing ids unless it's necessary.
+enum ObjectTypes {
+  Component = 'add-component',
+  Overlay = 'add-component-overlay',
+  DataBinding = 'add-component-data-binding',
+}
```

Comment:
Nit: Consider using union of string literals instead of enums. They offer the same effect and work in non-TypeScript projects. The TypeScript team has acknowledged it was a mistake to add functionality not supported by JavaScript and now only focuses on adding types to JavaScript.

Reply:
this enum is not supposed to be used by outside of scene composer so should be fine to not work in non-TS project.